### ISSUE TITLE
Additional / new logfile pattern for failed roundcube logins

### ISSUE THREAD

**Issue by kdslkdsaldsal** [wazuh/wazuh-ruleset#57]
I've noticed the following in the "errors" file of a current roundcube 1.2.5 installation:

```
[31-Aug-2017 13:50:32 +0200]: <foo> IMAP Error: Login failed for admin from 192.168.178.1. AUTHENTICATE PLAIN: Authentication failed. in /var/www/html/roundcube/program/lib/Roundcube/rcube_imap.php on line 193 (POST /roundcube/?_task=login&_action=login)
```

where it seems that https://github.com/wazuh/wazuh-ruleset/blob/master/decoders/0255-roundcube_decoders.xml doesn't match.

Cross-Ref: https://github.com/ossec/ossec-hids/issues/1245

**Comment by kdslkdsaldsal** [comment]
By default roundcube is also not logging into syslog but into an own file /var/www/html/roundcube/logs/errors (See: https://github.com/roundcube/roundcubemail/blob/1.2.5/config/defaults.inc.php#L70) 

Seems the "program_name" doesn't match in this case as well.

**Comment by snaow** [comment]
Hi @kdslkdsaldsal,

Current decoder is expecting different log headers, including the "_roundcube_" program_name:


> _Apr 10 22:45:20 hostname **roundcube**: [10-Apr-2009 22:45:20 -0500] IMAP Error: Authentication for username failed (LOGIN): "a001 NO Authentication`_

Your log line does not include any program name, you can create new decoder to match your syntax.

I created an example decoder for you:

```
<decoder name="roundcube">
  <prematch>^[\d\d-\w\w\w-\d\d\d\d \d\d:\d\d:\d\d +\d\d\d\d]</prematch>
</decoder>

<decoder name="roundcube-denied">
  <parent>roundcube</parent>
  <prematch>Login failed</prematch>
  <regex offset="after_prematch">^ for (\.+) from (\.+) </regex>
  <order>srcusr, srcip</order>
</decoder>

```

Log test output:

```
**Phase 1: Completed pre-decoding.
       full event: '[31-Aug-2017 13:50:32 +0200]: <foo> IMAP Error: Login failed for admin from 192.168.178.1. AUTHENTICATE PLAIN: Authentication failed. in /var/www/html/roundcube/program/lib/Roundcube/rcube_imap.php on line 193 (POST /roundcube/?_task=login&_action=login)'
       hostname: 'node2'
       program_name: '(null)'
       log: '[31-Aug-2017 13:50:32 +0200]: <foo> IMAP Error: Login failed for admin from 192.168.178.1. AUTHENTICATE PLAIN: Authentication failed. in /var/www/html/roundcube/program/lib/Roundcube/rcube_imap.php on line 193 (POST /roundcube/?_task=login&_action=login)'

**Phase 2: Completed decoding.
       decoder: 'roundcube'
       srcusr: 'admin'
       srcip: '192.168.178.1.'
```

Remember to add your new decoder into your custom decoder folders(etc/decoders) and exclude the current one in your ossec.conf file like this:

**ossec.conf**
```
<ruleset>
    <decoder_exclude>ruleset/decoders/0255-roundcube_decoders.xml</decoder_exclude>
 </ruleset>
```

Best regards,
Pedro.

**Comment by kdslkdsaldsal** [comment]
Hey, thanks for your reply. What about adding your ``<prematch>`` to the current roundcube decoder as roundcube isn't logging to syslog by default (which would be required for the "roundcube" program_name to match)?

**Comment by snaow** [comment]
Hi @kdslkdsaldsal,

Adding the prematch regex I created to the current decoder won't work.
As you said, _program_name_ is not been fulfilled in your _roundcube_ syntax, since _program_name_ is the very first thing we check, it won't continue if it does not match.

We could say that _program_name_ can't be used to decode your log line cause it is not matching for a default pre-decoder in Wazuh.

We have two options:

1. Adapt current decoder to match both regex, meaning, we should take out the program_name filter.
2. Create a new roundcube decoders (as I did before) which will match for non-syslog incoming events.

Best regards,
Pedro.
